modules/tensor_mechanics/tests/mohr_coulomb/tests

```diff
@@ -126,6 +133,13 @@
     rel_err = 1.0E-5
     abs_zero = 1.0E-3
   [../]
+  [./uni_axial1_small_strain]
+    type = 'CSVDiff'
+    input = 'uni_axial1_small_strain.i'
+    csvdiff = 'uni_axial1.csv'
+    rel_err = 1.0E-5
+    abs_zero = 1.0E-3
```

Comment:
Not necessarily.  It depends on many factors.  You could try a DirichletBC and see if that is better here... it depends on the particulars of the problem being solved.  Setting the "correct" value on the boundary using PresetBC _can_ cause the residual to spike _up_.  Explaining this in a text box gets tricky but let me try:

Think about solving Laplace's Equation div(grad(u)) = 0.  With a constant u=0 initial condition (initial guess) and BCs of 0 on the left and 1 on the right.

The initial residual will be zero everywhere except on the right boundary (a constant satisfies Laplace's equation and 0 on the left is satisfied by the initial condition).

If you use a DirichletBC then the right value will be "solved" up to 1 within the first timestep.  While it does that the constraints imposed by Laplace's equation remain satisfied... so the residual will go down toward zero in the whole domain.

However, if you use a PresetBC then the value of "u" will be set to 1 before the solve.  This will cause the residual to go to zero on the right side of the domain... which is good.  _However_... this will also cause the residual to spike in the nodes just one layer in from the right boundary.  Why is that?  Because the solution now has a discontinuity in the gradient where the flat (gradient=0) solution meets the right boundary (gradient~infinity).  This causes a huge misfit with Laplace's equation... which could dwarf the original boundary misfit and cause the residual to actually go _up_.

So why would we ever want to do that?  Because by doing this we've changed the nature of the solution misfit.  We took a problem that had a misfit in u-1=0 on a discrete set of nodes and turned it into one where the misfit is in Laplace's equation.  One thing this does is that it makes it much more amenable to multigrid... making it easier to solve for the response of the rest of the domain.

Anyway... I hope that helps a bit.  There is no "right" or "wrong" here... the choices are all correct in different circumstances...